src/core/a-assets.js

```diff
@@ -20,7 +20,7 @@ class AAssets extends ANode {
 
   connectedCallback () {
     // Defer if DOM is not ready.
-    if (document.readyState === 'loading') {
```

Comment:
what about if we check here for `document.readyState === 'complete'` instead?

Reply:
would need to call this.doConnectedCallback in the `DOMContentLoaded` event handler

Reply:
alternatively we can listen to [readystatechange](https://html.spec.whatwg.org/multipage/indices.html#event-readystatechange) and wait for document.readyState === 'complete' instead of listening to DOMContentLoaded

Reply:
>what about if we check for document.readyState !== 'complete' instead? I think that would wait until deferred scripts have already run

If we checked against readyState complete but still listened for DOMContentLoaded, that would open another gap in the loading order:


1. readyState: 'loading'
2. readyState: 'interactive'
3. Execute deferred scripts (including modules)
4. DOMContentLoaded
5. Execute DOMContentLoaded listeners
6. readyState: 'complete'

During step 5, the readyState would still be interactive, so any entities attached from inside a DOMContentLoaded listener would never initialize as they would exit early from their connectedCallback and stay forever waiting for the DOMContentLoaded event that has already passed

>would need to call this.doConnectedCallback in the DOMContentLoaded event handler

It does call through to `doConnectedCallback` as-is because the listener for the ANode static property is registered first and thus updates `ANode.DOMContentHasLoaded` before this comes back in via DOMContentLoaded listener. However, if you're open to updating all of these listeners to call `doConnectedCallback` I would love to make that change because it makes it much easier to subclass these when they call into doConnectedCallback via `this` rather than `prototype.doConnectedCallback` like how a-entity does currently

>alternatively we can listen to readystatechange and wait for document.readyState === 'complete' instead of listening to DOMContentLoaded

Yes I proposed this as one of the two possible solutions when opening #5228. So that would look like:

```
if (readyState !== 'complete') {
  // vital that the callback here is connectedCallback and not doConnectedCallback
  // to make sure we don't initialize early on the loading->interactive transition
  document.addEventListener('readystatechange', this.connectedCallback.bind(this));
  return;
}
```

Note this would conflict with the previous suggesting about changing the callback to doConnectedCallback. If we still wanted to do that as well, we'd need to add an event handler function that checked the current readyState to make sure it was complete before calling doConnectedCallback

Reply:
oops on second thought the `readystatechange` solution proposed above would not work; it could stack listeners and initialize twice. Would have to go with the creating a readystatechange handler

```
onReadyStateChange () {
  if (document.readyState === 'complete') {
    this.doConnectedCallback();
  }
}

connectedCallback () {
  // Defer if DOM is not ready.
  if (document.readyState !== 'complete') {
    document.addEventListener('readystatechange', this.onReadyStateChange.bind(this));
    return;
  }
  this.doConnectedCallback()
}
```

Reply:
> readystatechange

Thanks for all the info and patience. I think this solution looks simpler. Can we incorporate in the PR?

Reply:
Sounds good

Reply:
@dmarcos I did the readystatechange version, but it occurred to me during the work that this could be a breaking change. Anyone using `DOMContentLoaded` listeners of their own with A-Frame probably expects the scene and entities to be initialized (although I guess that would depend on whether they registered their listerens before or after A-Frame), but this would change it so scene and entities are never initialized during `DOMContentLoaded`. 

The unit tests may also be bearing this out, as the readystatechange version has one failure whereas this current version had none:

```
SUMMARY:
✔ 2302 tests completed
ℹ 16 tests skipped
✖ 1 test failed

FAILED TESTS:
    playSound
      ✖ plays sound if sound already playing when changing src
        Firefox 110.0 (Ubuntu 0.0.0)
      Timeout of 3000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```

So just wanted to check in if you still want to go this direction with readystatechange. If so, I'll dig into the unit test failure

Reply:
In A-Frame the 'loaded' event should be used. `DOMContentLoaded` is not very useful since won't guarantee the scene, assets and entities are ready to go.